### PR TITLE
Move TCL test unit/type/list to Go case

### LINKED ISSUE #816: Move TCL test unit/type/list to Go case

### PR DESCRIPTION
**Pull request by suica**
refer to #816

### CODE REVIEW

File: tests/gocase/unit/type/list/list_test.go [hunk 1]
```diff
@@ -39,6 +39,118 @@ var largeValue = map[string]string{
...
+	}
+
+	for typ, value := range largeValue {
+		t.Run(fmt.Sprintf("BLPOP, BRPOP: single existing list - %s", typ), func(t *testing.T) {
+			c := srv.NewTCPClient()
+			defer func() { require.NoError(t, c.Close()) }()
+			createList("blist", []string{"a", "b", value, "c", "d"})
+			res := rdb.LRange(ctx, "blist", 0, -1).Val()
+			fmt.Println(res)
+			require.NoError(t, c.Write("*3\r\n$5\r\nBLPOP\r\n$5\r\nblist\r\n$1\r\n1\r\n"))
```
Comment (suica) [inline tests/gocase/unit/type/list/list_test.go:148]: I used raw protocol here.

Comment (suica) [inline tests/gocase/unit/type/list/list_test.go:148]: @tisonkun Hi, I found out that in `list.tcl`, some cases use `set rd [redis_deferring_client]` as a deferring client.

I wonder how to use a "deferring client" in Go.

---

The study I have done:
- I have tried `NewTCPClient`, and although it has similar APIs (such as `ReadLine`), but it turns out to be that I have to write raw Redis protocol to communicate with the server. Actually it works, but it seems to be a little bit awkward... Any way, I have written a little piece of code at [here](https://github.com/apache/incubator-kvrocks/pull/866/files#diff-6df089ad841b180ccd3a65619ee756b051f26ed20ea3cc984feb985c41c9da02). Please comment on this if I did anything wrong :)
- I also have learned how you dealt with the deferring client in #854 and #810, and it seems to be we still need to write raw protocol in code.

**Comment by tisonkun** [comment]
@ShooterIT @git-hulk I'm unfamiliar how `redis_deferring_client` works. Could you give some background here?

@suica for raw client, in #888 I write a `WriteArgs` method for easily writing normal commands.

**Comment by git-hulk** [comment]
> @ShooterIT @git-hulk I'm unfamiliar how `redis_deferring_client` works. Could you give some background here?
> 
> @suica for raw client, in #888 I write a `WriteArgs` method for easily writing normal commands.

`redis_deferring_client` will NOT read the response after sending commands, so we can send many commands then reading them together. Can see https://github.com/apache/incubator-kvrocks/blob/fca9e6962be5e85ad659545d94b5dafa4aec1325/tests/tcl/tests/support/redis.tcl#L140

**Comment by tisonkun** [comment]
Closed as inactive. Code conflict by the way. @suica you can resubmit the patch when it gets completed.

**Pull request closed without merging**